src/infrastructure/databaseService.ts

```diff
@@ -0,0 +1,54 @@
+import mysqlp, { FieldPacket, RowDataPacket, Connection } from 'mysql2/promise';
+import SecretsManagerServiceInterface from '../interfaces/SecretsManagerService';
+import DatabaseServiceInterface from '../interfaces/DatabaseService';
+
+export default class DatabaseService implements DatabaseServiceInterface {
+  connection: Connection;
+
+  constructor(connection: Connection) {
+    this.connection = connection;
+  }
+
+  async get(query: string, params: string[] | undefined): Promise<[RowDataPacket[], FieldPacket[]]> {
+    try {
+      console.info(`Executing query ${query} with params ${params.join(', ')}`);
+      const result = await this.connection.execute<RowDataPacket[]>(query, params);
+
+      return result;
+    } catch (e) {
+      // Type checking because the type of e can't be specified in the params
+      if (e instanceof Error) {
+        e.message = `Database error: ${e.message}`;
+      }
+
+      throw e;
+    }
+  }
+
+  public static async build(
+    secretsManager: SecretsManagerServiceInterface,
+    mysql: typeof mysqlp,
+  ): Promise<DatabaseServiceInterface> {
+    const dbConnectionDetailsString = await secretsManager.getSecret(process.env.SECRET);
```

Comment:
We might want to update the README, env and .env with this value otherwise the service will fail locally I think.
The DevOps will probabky need to add that in AWS too

Reply:
It's already in AWS, I'll add it to the others.